towhee/engine/operator_io/reader.py

```diff
@@ -90,18 +89,35 @@ def read(self) -> Tuple[Dict[str, any], Tuple]:
         if self._close:
             raise StopIteration
 
+        ret = {}
+        origin_data = ()
+        bitmap = list(range(self._iters_count))
+
         with self._lock:
-            data = next(self._iter)
             if self._close:
                 raise StopIteration
 
-            if not data:
-                return {}, ()
-            return self._to_op_inputs(data), data
+            #TODO: filip-halt
+            #Take advantage of iterator block.
+            while len(bitmap) > 0:
+                remove_index = set()
+                for bit_index, iter_index in enumerate(bitmap):
+                    # self._iters[_][0] is the iterator
+                    # self._iters[_][1] are the columns for that iterator
+                    data = next(self._iters[iter_index][0])
+                    if data is not None:
+                        for (key, index) in self._iters[iter_index][1]:
+                            ret[key] = data[index].value
+                        origin_data += data
+                        remove_index.add(bit_index)
+                bitmap = [i for j, i in enumerate(bitmap) if j not in remove_index]
+            return ret, origin_data
```

Comment:
This reader blocks by looping to get the remaining values. Should it rather use the new block feature in the iterator?